mergify_engine/clients/github.py

```diff
@@ -58,9 +62,15 @@ class GithubInstallationClient(common.BaseClient):
     def __init__(self, owner, repo, installation_id=None):
         self.owner = owner
         self.repo = repo
-        self.installation_id = installation_id or github_app.get_client().get_installation_id(
-            owner, repo
-        )
+        self.installation_id = github_app.get_client().get_installation_id(owner, repo)
+        if self.installation_id != installation_id:
+            LOG.error(
+                "installation id for repository diff from event installation id",
+                gh_owner=owner,
+                gh_repo=repo,
+                installation_id=self.installation_id,
+                expected_installation_id=installation_id,
+            )
```

Comment:
this will end up in sentry, do we want that?

Reply:
Oh yes ! We should never happen, but I want check if I'm right.